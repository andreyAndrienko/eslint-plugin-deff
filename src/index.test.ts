import * as child_process from "child_process";
import { mocked } from "ts-jest/utils";

jest.mock("child_process");
const mockedChildProcess = mocked(child_process, true);
mockedChildProcess.execSync.mockReturnValue(Buffer.from("line1\nline2\nline3"));

import { configs, processors } from "./index";

describe("plugin", () => {
  it("should match expected export", () => {
    expect(configs).toMatchSnapshot();
    expect(processors).toMatchSnapshot();
  });
});
