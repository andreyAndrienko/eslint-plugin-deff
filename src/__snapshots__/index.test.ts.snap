// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin should match expected export 1`] = `
Object {
  "diff": Object {
    "ignorePatterns": Array [],
    "overrides": Array [
      Object {
        "files": Array [
          "*",
        ],
        "processor": "diff/diff",
      },
    ],
    "plugins": Array [
      "diff",
    ],
  },
  "staged": Object {
    "ignorePatterns": Array [],
    "overrides": Array [
      Object {
        "files": Array [
          "*",
        ],
        "processor": "diff/staged",
      },
    ],
    "plugins": Array [
      "diff",
    ],
  },
}
`;

exports[`plugin should match expected export 2`] = `
Object {
  "diff": Object {
    "postprocess": [Function],
    "supportsAutofix": true,
  },
  "staged": Object {
    "postprocess": [Function],
    "supportsAutofix": true,
  },
}
`;
